packages/react/src/components/AmplifyProvider/__tests__/AmplifyProvider.test.tsx

```diff
@@ -37,4 +38,48 @@ describe('AmplifyProvider', () => {
       ''
     );
   });
+
+  it('filters out XSS attacks which attempt to escape the CSS context', async () => {
+    const name = 'maliciousTheme';
+    const xss = `</style><script>alert('XSS')</script>`;
+    const maliciousTheme: Theme = {
+      name,
+      tokens: {
+        colors: {
+          font: { primary: { value: 'pink' }, secondary: { value: xss } },
+        },
+      },
+    };
+
+    const { container } = render(
+      <AmplifyProvider theme={maliciousTheme}>
+        <App />
+      </AmplifyProvider>
+    );
+
+    const styleTag = container.querySelector(`#amplify-theme-${name}`);
+    expect(styleTag).toBe(null);
+  });
+
+  it('filters out XSS attacks which attempt to escape the CSS context using whitespace characters', async () => {
+    const name = 'maliciousTheme';
+    const xss = `</style\r ><script>alert('XSS')</script>`;
```

Comment:
Can we add another test like this?( backticks preserve carriage returns)
```suggestion
    const xss = `</style
    ><script>alert('XSS')</script>`;
```